hyperspy/io_plugins/protochips.py

```diff
@@ -0,0 +1,217 @@
+# be necessary to mention the other reader in this message (and to add an
+# argument in the load function to specify the correct reader)
+invalid_file_error = "The Protochips csv reader can't import the file, please"\
+    " make sure, this is a valid Protochips log file."
+
+
+def _bad_file():
+    raise IOError(invalid_file_error)
+
+
+def file_reader(filename, *args, **kwds):
+    csv_file = ProtochipsCSV(filename)
+    return _protochips_log_reader(csv_file)
+
+
+def _protochips_log_reader(csv_file):
+    csvs = []
+    for key in csv_file.logged_quantity_name_list:
+        try:
+            csvs.append(csv_file.get_dictionary(key))
+        except:
+            _bad_file()
+    return csvs
+
+
+class ProtochipsCSV(object):
+
+    def __init__(self, filename, header_line_number=10):
+        self.filename = filename
+        self._parse_header(header_line_number)
+        self._read_data(header_line_number)
+        self.calibration_file = None
+
+    def _parse_header(self, header_line_number):
+        self.raw_header = self._read_header(header_line_number)
+        self.column_name = self._read_column_name()
+        self._check_if_protochips_csv_file()
+        self._read_all_metadata_header()
+        self.logged_quantity_name_list = self.column_name[2:]
+
+    def _check_if_protochips_csv_file(self):
+        if 'Time' in self.column_name and 'Notes' in self.column_name and len(
+                self.column_name) >= 3:
+            pass
+        else:
+            _bad_file()
+
+    def get_dictionary(self, quantity):
+        return {'data': self._data_dictionary[quantity],
+                'axes': self._get_axes(),
+                'metadata': self._get_metadata(quantity),
+                'original_metadata': {'Protochips_header':
+                                      self._get_original_metadata()}}
+
+    def _get_original_metadata(self):
+        d = {'Start time': self.start_datetime}
+        d['Time units'] = self.time_units
+        for quantity in self.logged_quantity_name_list:
+            d['%s_units' % quantity] = self._parse_quantity_units(quantity)
+        d['User'] = self.user
+        d['Calibration file name'] = self._parse_calibration_file()
+        d['Time axis'] = self._get_metadata_time_axis()
+        return d
+
+    def _get_metadata(self, quantity):
+        return {'General': {'original_filename': os.path.split(self.filename)[1],
+                            'title': '%s (%s)' % (quantity,
+                                                  self._parse_quantity_units(quantity)),
+                            'authors': self.user,
+                            'date': self.start_datetime.date(),
+                            'time': self.start_datetime.time(),
+                            'notes': self._parse_notes()},
+                "Signal": {'signal_type': '',
+                           'quantity': self._parse_quantity_name(quantity),
```

Comment:
Oh, and quantity must also be added to the metadata specification. Note that all quantities with units include the units in the quantity string e.g. "beam energy (keV)". I think that it'll be good to be consistent and do the same for quantity.